src/geometry/manifold/partition_of_unity.lean

```diff
@@ -193,4 +302,106 @@ fs.mem_ext_chart_at_source_of_eq_one (fs.apply_ind x hx)
+  fs.to_smooth_partition_of_unity i x = 0 :=
+fs.to_bump_covering.to_partition_of_unity_zero_of_zero h
+
+lemma support_to_smooth_partition_of_unity_subset (i : fs.ι) :
+  support (fs.to_smooth_partition_of_unity i) ⊆ support (fs i) :=
+fs.to_bump_covering.support_to_partition_of_unity_subset i
+
+lemma is_subordinate.to_smooth_partition_of_unity {f : smooth_bump_covering I M s} {U : M → set M}
+  (h : f.is_subordinate U) :
+  f.to_smooth_partition_of_unity.is_subordinate (λ i, U (f.c i)) :=
+h.to_bump_covering.to_partition_of_unity
+
+lemma sum_to_smooth_partition_of_unity_eq (x : M) :
+  ∑ᶠ i, fs.to_smooth_partition_of_unity i x = 1 - ∏ᶠ i, (1 - fs i x) :=
+fs.to_bump_covering.sum_to_partition_of_unity_eq x
+
 end smooth_bump_covering
+
+variable (I)
+
+/-- Given two disjoint closed sets in a Hausdorff σ-compact finite dimensional manifold, there
+exists an infinitely smooth function that is equal to `0` on one of them and is equal to one on the
+other. -/
+lemma exists_smooth_zero_one_of_closed [t2_space M] [sigma_compact_space M] {s t : set M}
+  (hs : is_closed s) (ht : is_closed t) (hd : disjoint s t) :
+  ∃ f : C^∞⟮I, M; 𝓘(ℝ), ℝ⟯, eq_on f 0 s ∧ eq_on f 1 t ∧ ∀ x, f x ∈ Icc (0 : ℝ) 1 :=
+begin
+  have : ∀ x ∈ t, sᶜ ∈ 𝓝 x, from λ x hx, hs.is_open_compl.mem_nhds (disjoint_right.1 hd hx),
+  rcases smooth_bump_covering.exists_is_subordinate I ht this with ⟨f, hf⟩,
+  set g := f.to_smooth_partition_of_unity,
+  refine ⟨⟨_, g.smooth_sum⟩, λ x hx, _, λ x, g.sum_eq_one, λ x, ⟨g.sum_nonneg x, g.sum_le_one x⟩⟩,
+  suffices : ∀ i, g i x = 0,
+    by simp only [this, times_cont_mdiff_map.coe_fn_mk, finsum_zero, pi.zero_apply],
+  refine λ i, f.to_smooth_partition_of_unity_zero_of_zero _,
+  exact nmem_support.1 (subset_compl_comm.1 (hf.support_subset i) hx)
+end
+
+variable {I}
+
+namespace smooth_partition_of_unity
+
+variables [t2_space M] [sigma_compact_space M]
+
+/-- If `X` is a paracompact normal topological space and `U` is an open covering of a closed set
+`s`, then there exists a `bump_covering ι X s` that is subordinate to `U`. -/
+lemma exists_is_subordinate {s : set M} (hs : is_closed s) (U : ι → set M) (ho : ∀ i, is_open (U i))
+  (hU : s ⊆ ⋃ i, U i) :
+  ∃ f : smooth_partition_of_unity ι I M s, f.is_subordinate U :=
+begin
+  haveI : locally_compact_space H := I.locally_compact,
+  haveI : locally_compact_space M := charted_space.locally_compact H,
+  haveI : normal_space M := normal_of_paracompact_t2,
+  rcases bump_covering.exists_is_subordinate_of_prop (smooth I 𝓘(ℝ)) _ hs U ho hU
+    with ⟨f, hf, hfU⟩,
+  { exact ⟨f.to_smooth_partition_of_unity hf, hfU.to_smooth_partition_of_unity hf⟩ },
+  { intros s t hs ht hd,
+    rcases exists_smooth_zero_one_of_closed I hs ht hd with ⟨f, hf⟩,
+    exact ⟨f, f.smooth, hf⟩ }
+end
+
+instance {s : set M} [is_closed s] :
+  inhabited (smooth_partition_of_unity s I M s) :=
+⟨(exists_is_subordinate ‹_› (λ _ : s, univ) (λ _, is_open_univ)
+  (λ x hx, mem_Union.2 ⟨⟨x, hx⟩, mem_univ x⟩)).some⟩
```

Comment:
This one is really weird. Why not use the constant function equal to 1?

Reply:
Now I'm using `bump_covering.single.to_smooth_partition_of_unity`. Is it better?